core/src/test/scala/unit/kafka/log/ProducerStateManagerTest.scala

```diff
@@ -321,6 +321,28 @@ class ProducerStateManagerTest extends JUnitSuite {
   }
 
   @Test
+  def testAcceptAppendWithoutProducerStateOnReplica(): Unit = {
```

Comment:
Could we have a similar test case which has a gap in the sequence numbers and verifies that they are accepted when `isFromClient` is false?

Reply:
ack. Added this test case.